packages/react/src/components/linkbox/Linkbox.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useRef } from 'react';
+
+// import core base styles
+import 'hds-core';
+import styles from './Linkbox.module.scss';
+import { IconArrowRight, IconLinkExternal } from '../../icons';
+import classNames from '../../utils/classNames';
+
+export type LinkboxProps = Omit<React.ComponentPropsWithoutRef<'a'>, 'tabIndex'> & {
+  /**
+   * Aria-label of the linkbox component.
+   */
+  ariaLabel: string;
+  /**
+   * Boolean indicating for external link that takes user to an entirely new web site. Defaults to false.
+   */
+  external?: boolean;
+  /**
+   * Optional heading of the linkbox.
+   */
+  heading?: string;
+  /**
+   * Heading level for linkbox title. Defaults to 2.
+   */
+  headingAriaLevel?: number;
+  /**
+   * The image props passed to the linkbox image. Image variant requires prop withImg set to true.
+   */
+  imgProps?: React.ComponentPropsWithoutRef<'img'>;
+  /**
+   * Boolean for the variant with no background color. Changes paddings also. Defaults to false.
+   */
+  noBackground?: boolean;
+  /**
+   * Aria label for opening link in an external domain. Active when external prop is true.
+   */
+  openInExternalDomainAriaLabel?: string;
+  /**
+   * Boolean indicating whether the link will open in new tab or not. Defaults to false.
+   */
+  openInNewTab?: boolean;
+  /**
+   * Aria label for opening link in a new tab
+   */
+  openInNewTabAriaLabel?: string;
+  /**
+   * Optional text of the linkbox.
+   */
+  text?: string;
+  /**
+   * Boolean indicating whether the Linkbox contains a border or not. Defaults to false.
+   */
+  withBorder?: boolean;
+  /**
+   * Boolean indicating whether the Linkbox contains a hero image or not. Defaults to false.
+   */
+  withImg?: boolean;
+  /**
+   * Size variant for the linkbox when accompanied with an img. Defaults to 'small'
+   */
+  withImgSize?: 'small' | 'medium' | 'large';
```

Comment:
This might be confusing to developer. The size does not affect to image size in any way. It only affects to font sizes and paddings that are used for heading and text of the linkbox. Currently the size variants are only used with images. So regular linkbox does not have those. Therefore the name of this prop is linked to the img. If we would allow size variants for also the version without image, then this could be more easily named as: 'ContentSize' or something like that 🤔

Reply:
- [ ] Fix by setting style to affect every variant